Marlin/Marlin_main.cpp

```diff
@@ -596,6 +596,43 @@ void setup_powerhold() {
   #endif
 }
 
+/*
+  Enable 24V to fans, E-reg, Cartridge Holder, Cartridges
+  if no short circuit is detected on line
+  ~200us, blocking time
+  Requires serial to be enabled to generate error condition
+*/
+void enable_power_supply() {
+  // Turn on FAN first so that 24V does not have a path to 24V_SW through the Rambo. TODO: Disconnect these two circuits.
+  pinMode(FAN_CHASSIS_TOP_PIN, OUTPUT);
+  digitalWrite(FAN_CHASSIS_TOP_PIN, HIGH);
+  delay(6);  // 40ms delay for line to settle. NOT NEEDED. Fuse set to 65ms startup delay by default.
+  
+  analogRead(PS_MONITOR_PIN);  // Throw out first reading
+  int V_Monitor_Result = analogRead(PS_MONITOR_PIN);
+  if (V_Monitor_Result > PS_ENABLE_LOWER_LIMIT)   // Test for 24V short to gnd
+  {
+    if (V_Monitor_Result <= PS_ENABLE_UPPER_LIMIT) // Test for 24V short to 5V
+    {
+      digitalWrite(PS_FORCE_ON_LL,LOW); // Pull Power Supply enable low to force on
+      pinMode(PS_FORCE_ON_LL, OUTPUT);  //
+      delayMicroseconds(100);           // 100us minimum delay to register on
+      pinMode(PS_FORCE_ON_LL, INPUT);   // Release Power Supply enable to allow current limiting
+    }
+    else 
+    {
+      // TODO: Add to error handling protocol
+      // TODO: Figure out if coming up from power-down state or reset condition
+      //SERIAL_ECHOLN("ERROR TURNING ON 24V POWER. Detected 24V/5V short... PLEASE CHECK HARDWARE...");
+    }
+  }  
+  else // Handle error case (short circuit or open circuit?)
+  {
+    // TODO: Add to error handling protocol
+    SERIAL_ECHOLN("ERROR TURNING ON 24V POWER. Detected 24V/gnd short... PLEASE CHECK HARDWARE...");  
```

Comment:
Yeah, bad things will likely happen if we don't enable downstream power and then try to run the printer (steppers would still work fine, for instance). It will be fairly obvious if you are at the printer (lights won't turn on, fans, etc.). I was counting on the user to notice this when they power up for now. 

This code does NOT check the line at run-time to see if we have hit a current-limit condition, only at startup.

This is something I was planning on adding later, but we can wait and push this all at once. We will just have to branch code betweem Gen3C and Gen3C.2 when we do.